src/rpc/rpccategory.cpp

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <rpc/rpccategory.h>
+
+namespace rpccategory
+{
+
+std::string Label (const RPCCategory category);
+
+std::string Label (const RPCCategory category)
+{
+    #ifdef HANDLE_CASE_RETURN
+    #error "HANDLE_CASE_RETURN already defined"
+    #else
+    #define HANDLE_CASE_RETURN(classname,member) case classname::member: return #member
+    #endif
+
+    switch (category)
+    {
+            HANDLE_CASE_RETURN (RPCCategory, blockchain);
+            HANDLE_CASE_RETURN (RPCCategory, control);
+            HANDLE_CASE_RETURN (RPCCategory, generating);
+            HANDLE_CASE_RETURN (RPCCategory, hidden);
+            HANDLE_CASE_RETURN (RPCCategory, mining);
+            HANDLE_CASE_RETURN (RPCCategory, network);
+            HANDLE_CASE_RETURN (RPCCategory, rawtransactions);
+            HANDLE_CASE_RETURN (RPCCategory, util);
+            HANDLE_CASE_RETURN (RPCCategory, wallet);
+            HANDLE_CASE_RETURN (RPCCategory, zmq);
+            HANDLE_CASE_RETURN (RPCCategory, test);
+            // if you are missing a case, you'll have a warning here
+    }
+    #undef HANDLE_CASE_RETURN
```

Comment:
I think we don't need to undef this

Reply:
ok right now, it's not necessary, but i meant to define a local MACRO: define and use as you like, and undefined when you don't need anymore.